atomix/cluster/src/test/java/io/atomix/raft/roles/PassiveRoleTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.roles;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.atomix.raft.impl.RaftContext;
+import io.atomix.raft.metrics.RaftReplicationMetrics;
+import io.atomix.raft.protocol.AppendRequest;
+import io.atomix.raft.protocol.AppendResponse;
+import io.atomix.raft.storage.RaftStorage;
+import io.atomix.raft.storage.log.RaftLog;
+import io.atomix.raft.storage.log.RaftLogWriter;
+import io.atomix.raft.storage.log.entry.RaftLogEntry;
+import io.atomix.raft.zeebe.ZeebeEntry;
+import io.atomix.storage.journal.Indexed;
+import io.atomix.utils.serializer.NamespaceImpl;
+import io.atomix.utils.serializer.NamespaceImpl.Builder;
+import io.atomix.utils.serializer.Namespaces;
+import io.zeebe.snapshots.raft.PersistedSnapshot;
+import io.zeebe.snapshots.raft.ReceivableSnapshotStore;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class PassiveRoleTest {
```

Comment:
We are missing a test to verify what happens when checksum is invalid.

Reply:
That's not checked in the PassiveRole. That test is in the JournalSegmentWriterTest